volatility3/framework/plugins/windows/crashinfo.py

```diff
@@ -0,0 +1,83 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import datetime
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints, conversion
+from volatility3.framework.objects import utility
+
+vollog = logging.getLogger(__name__)
+
+class Crashinfo(interfaces.plugins.PluginInterface):
+    _required_framework_version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]), 
+            ]
+        
+    def _generator(self, layer):
+        header = layer.get_header()
+        uptime = datetime.timedelta(microseconds=int(header.SystemUpTime) / 10)
+
+        if header.DumpType == 0x1:
+            dump_type = "Full Dump (0x1)"
+        elif header.DumpType == 0x5:
+            dump_type = "Bitmap Dump (0x5)"
+        else:
+            # this should never happen since the crash layer only accepts 0x1 and 0x5
+            dump_type = "Unknown/Unsupported ({:#x})".format(header.DumpType)
+
+        if header.DumpType == 0x5:
+            summary_header = layer.get_summary_header()
+            bitmap_header_size = format_hints.Hex(summary_header.HeaderSize)
+            bitmap_size = format_hints.Hex(summary_header.BitmapSize)
+            bitmap_pages = format_hints.Hex(summary_header.Pages)
+        else:
+            bitmap_header_size = bitmap_size = bitmap_pages = renderers.NotApplicableValue()
+
+        yield(0, (utility.array_to_string(header.Signature),
+                  header.MajorVersion,
+                  header.MinorVersion,
+                  format_hints.Hex(header.DirectoryTableBase),
+                  format_hints.Hex(header.PfnDataBase),
+                  format_hints.Hex(header.PsLoadedModuleList),
+                  format_hints.Hex(header.PsActiveProcessHead),
+                  header.MachineImageType,
+                  header.NumberProcessors,
+                  format_hints.Hex(header.KdDebuggerDataBlock),
+                  dump_type,
+                  str(uptime),
+                  utility.array_to_string(header.Comment),
+                  conversion.wintime_to_datetime(header.SystemTime),
+                  bitmap_header_size,
+                  bitmap_size,
+                  bitmap_pages,
+                  ))
+
+    def run(self):
+        layer = self._context.layers[self.config['primary.memory_layer']]
```

Comment:
So this plugin won't run against a crashdump layer unless it also managed to succesfully stack an intel layer on top of it.  With this particular format, that's exceptionally likely, but it would be better as example code, if it took whatever layer it was given and drilled down until it found a CrashDump layer.  This also avoids issues with calls to crashdump-specific methods (like the new `get_header`).  Calling those on an image that formed an intel layer, but whose base isn't a crashdump will throw some very ugly exceptions, which we should be coping with...

Reply:
Fixed in b15e69e